R/permutation.R

```diff
@@ -0,0 +1,703 @@
+
+
+#' Run augsynth letting each unit be the treated unit.  This
+#' implements the abadie placebo test (e.g., from tobacco paper) but
+#' with the augsynth package.
+#'
+#' @param tx.id Name/ID of treatment group (not stored in ascm)
+#'
+#' @return Dataframe With each row corresponding to a unit (including
+#'   the original treated unit) and each column corresponding to a
+#'   time point.  Each entry is the estimated impact for that unit at
+#'   that time, after fitting augsynth on that unit as the tx unit and
+#'   the other units as possible controls.
+#'
+#' @noRd
+get_placebo_gaps = function( ascm, att = TRUE ) {
+
+    tx_id <- ascm$trt_unit
+    wide_data <- ascm$data
+    synth_data <- ascm$data$synth_data
+    Z <- wide_data$Z
+    control_ids = which( wide_data$trt == 0 )
+    all_ids = 1:length(wide_data$trt)
+    t_final = length( wide_data$time )
+
+    ests <- vapply(all_ids, function(i) {
+        new_data <- swap_treat_unit(wide_data, Z, i)
+        new_ascm <- do.call(augsynth:::fit_augsynth_internal,
+                            c(list(wide = new_data$wide,
+                                   synth_data = new_data$synth_data,
+                                   Z = new_data$Z,
+                                   progfunc = ascm$progfunc, scm = ascm$scm,
+                                   fixedeff = ascm$fixedeff),
+                              ascm$extra_args))
+        est <- predict(new_ascm, att = att )
+        est
+    }, numeric(t_final) )
+
+    # Verify we recover the original treated unit by seeing if the
+    # estimates match our original fit model's estimates
+    dim( ests )
+    pds = as.numeric( predict( ascm, att = att ) )
+    pds
+    if( !all( round( ests[ , which( wide_data$trt == 1 ) ] - pds, digits=4 ) == 0 ) ) {
+        stop( "Two versions of estimated impacts do not correspond.  Serious error.  Please contact package maintainers." )
+    }
+
```

Comment:
When has this happened in testing? Is there a minimum reproducible example? If this is happening there should be a unit test against this and it should be fixed.